code/game/objects/items/food/pastries.dm

```diff
@@ -802,14 +817,19 @@
 	add_overlay(pancake_visual)
 	update_icon()
 
+/obj/item/food/pancakes/proc/On_Consume(eater, feeder)
+	if(istype(loc, /obj/item/food/pancakes)) //this is a pancake on the stack
+		var/obj/item/food/pancakes/stack = loc
+		addtimer(CALLBACK(stack, .update_icon),1) //remove the stack's last pancake we ate.
```

Comment:
why the timer

Reply:
We talked it out in discord, for maintainers coming by